cvmfs/magic_xattr.cc

```diff
@@ -222,28 +227,75 @@ std::string CatalogCountersMagicXattr::GetValue() {
+  chunk_list_ = "ERROR: No chunks! ";
   if (!dirent_->IsRegular()) {
+    chunk_list_ += "Not a regular file\n";
     return false;
   }
+
   if (dirent_->IsChunkedFile()) {
     FileChunkList chunks;
     if (!xattr_mgr_->mount_point()->catalog_mgr()
                      ->ListFileChunks(path_, dirent_->hash_algorithm(), &chunks)
         || chunks.IsEmpty())
     {
+      chunk_list_ += "File marked as chunked, but no chunks found\n";
       LogCvmfs(kLogCvmfs, kLogDebug | kLogSyslogErr, "file %s is marked as "
                 "'chunked', but no chunks found.", path_.c_str());
       return false;
     } else {
-      for (size_t i = 0; i < chunks.size(); ++i) {
+      const size_t kB_limit = 64 * 1024;
+      const size_t header_size = 120;
+      const size_t max_digits_offset =
+                      StringifyInt(chunks.At(chunks.size()-1).offset()).size();
+
+      // entry_size is calculated to find out how many entries we can show per
+      // page. It depends on the file itself, depending how many characters are
+      // needed to represent the largest number.
+      // This is some pessimisitic estimation.
+      //
+      // The number of characters per line is calculated the following:
+      // characters per entry =
+      //        2 * num pages  (can have up to 999 pages)
+      // +      maximum chunk size
+      // +      hash (is constant value for all hashes)
+      // +      max offset within file
+      // +      max offset within file (this should be the max size of any
+      //                             chunk, but for simplicity we use this here.
+      //                             this should be for sure larger.)
+      const size_t entry_size = (3ul * 2) +
+                                StringifyUint(chunks.size()).size() +
+                                chunks.At(0).content_hash().ToString().size() +
+                                max_digits_offset + max_digits_offset;
+      const size_t entries_per_page = (kB_limit - header_size) / entry_size;
+
+      const size_t num_pages = chunks.size() / entries_per_page +
+                              ((chunks.size() % entries_per_page != 0) ? 1 : 0);
+
+      chunk_list_  = "Access different pages with 'chunk_list_<page>' ";
+      chunk_list_ += "( " + StringifyInt(num_pages);
+      chunk_list_ += num_pages > 1 ? " pages )\n" : " page )\n";
+
+      chunk_list_ += "max_pages,curr_page,idx,hash,offset,size\n";
+
+      attr_page_num = attr_page_num == 0 ? 1 : attr_page_num;
+      size_t attr_page_num_fixed = static_cast<size_t>(attr_page_num) - 1;
+      size_t start_entry = entries_per_page * attr_page_num_fixed;
+      for (size_t i = start_entry;
+                  i < chunks.size() && i < start_entry + entries_per_page;
+                  ++i) {
+        chunk_list_ += StringifyUint(num_pages) + ",";
+        chunk_list_ += StringifyInt(attr_page_num) + ",";
+        chunk_list_ += StringifyUint(i) + ",";
```

Comment:
I don't think we need this level of sophistication. Here is my suggestion for a simpler way:
- Let's not show the page number and number of pages in the result, i.e. let's not modify the payload (it's ok to keep the information about pagination in the header)
- Let's create a vector of strings for the attribute value. Every element of the vector corresponds to a page. We add line by line to the string of the current page. If that page gets bigger than the limit, we append a new element to the vector. At the end (or when we reach the page number), we return that vector element.

Actually, I think we should prepare this vector entirely in `PrepareValueFenced()` and then return the vector element that corresponds to the page in `GetValue()`. This may come as a follow-up refactoring.